src/main/java/core/basesyntax/strategy/impl/PurchaseHandler.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void execute(String fruit, int quantity) {
+        int storedAmount = Storage.STORAGE.get(fruit);
```

Comment:
```suggestion
        int storedAmount = Storage.STORAGE.getOrDefault(fruit, 0);
```
if there is no such key in map null will be returned. And you got NPE (int primitive and cast null to it will lead to npe). And the same for ReturnHandler, SupplyHandler =)